platform/src/main/java/org/stellar/anchor/platform/callback/PlatformIntegrationHelper.java

```diff
@@ -50,34 +50,31 @@ public static AnchorException httpError(String responseContent, int responseCode
         "Error returned from the Anchor Backend.\nresponseCode={}\nContent={}",
         responseCode,
         responseContent);
-    ErrorResponse errorResponse;
+    ErrorResponse errorResponse = null;
     try {
       errorResponse = gson.fromJson(responseContent, ErrorResponse.class);
-    } catch (Exception e) { // cannot read body from response
-      return new ServerErrorException("internal server error", e);
-    }
-
-    // Handle 422
-    String errorMessage;
-    if (responseCode == HttpStatus.UNPROCESSABLE_ENTITY.value()) {
-      errorMessage =
-          (errorResponse != null)
-              ? errorResponse.getError()
-              : HttpStatus.BAD_REQUEST.getReasonPhrase();
-      return new BadRequestException(errorMessage);
+    } catch (Exception e) {
+      // cannot read body from response
+      Log.warn("Failed to parse responseContent to an ErrorResponse object.");
     }
 
-    errorMessage =
+    String errorMessage =
         (errorResponse != null)
             ? errorResponse.getError()
-            : HttpStatus.valueOf(responseCode).getReasonPhrase();
-    if (responseCode == HttpStatus.BAD_REQUEST.value()) {
-      return new BadRequestException(errorMessage);
-    } else if (responseCode == HttpStatus.NOT_FOUND.value()) {
-      return new NotFoundException(errorMessage);
+            : (responseCode == 422)
+                ? HttpStatus.BAD_REQUEST.getReasonPhrase()
+                : HttpStatus.valueOf(responseCode).getReasonPhrase();
+
+    switch (responseCode) {
+      case 422:
+      case 400:
+        return new BadRequestException(errorMessage);
+      case 404:
+        return new NotFoundException(errorMessage);
+      default:
+        Log.errorF("Unsupported status code {}.", responseCode);
+        return new ServerErrorException("internal server error");
```

Comment:
Can we use the HttpStatus code?

```suggestion
    switch (responseCode) {
      case UNPROCESSABLE_ENTITY:
      case BAD_REQUEST:
        return new BadRequestException(errorMessage);
      case NOT_FOUND:
        return new NotFoundException(errorMessage);
      default:
        Log.errorF("Unsupported status code {}.", responseCode);
        return new ServerErrorException("internal server error");
```

Reply:
Done in https://github.com/stellar/java-stellar-anchor-sdk/pull/612/commits/e779fbaf4b5fba524a0698a8207a050d885d86e4.